src/metabase/events/driver_notifications.clj

```diff
@@ -0,0 +1,41 @@
+(ns metabase.events.driver-notifications
+  (:require [clojure.core.async :as async]
+            [clojure.tools.logging :as log]
+            [metabase.db :as db]
+            [metabase.driver :as driver]
+            [metabase.events :as events]
+            [metabase.models.database :refer [Database]]))
+
+
+(def ^:const driver-notifications-topics
+  "The `Set` of event topics which are subscribed to for use in driver notifications."
+  #{:database-update})
+
+(def ^:private driver-notifications-channel
+  "Channel for receiving event notifications we want to subscribe to for driver notifications events."
+  (async/chan))
+
+
+;;; ## ---------------------------------------- EVENT PROCESSING ----------------------------------------
+
+
+(defn process-driver-notifications-event
+  "Handle processing for a single event notification received on the driver-notifications-channel"
+  [driver-notifications-event]
+  ;; try/catch here to prevent individual topic processing exceptions from bubbling up.  better to handle them here.
+  (try
+    (when-let [{topic :topic database :item} driver-notifications-event]
```

Comment:
this is nitpicky but I would prefer putting the `try` inside of the `when-let` instead of vice versa. That way if the `when-let` doesn't execute it's body we avoid the overhead of a `try-catch` block. Otherwise we'd be wasting a few CPU cycles